pkgs/development/haskell-modules/generic-builder.nix

```
@@ -541,7 +607,16 @@ let
       export GHC_PACKAGE_PATH="''${NIX_GHC_PACKAGE_PATH_FOR_TEST}"
     fi
 
-    exec "$@"
+    ${
+      let
+        runner =
+          if (isCross && crossSupport.canCheck) then
+            (if stdenv.hostPlatform.isGhcjs then "node" else crossSupport.emulator)
+          else
+            "exec";
+      in
+      runner + " " + ''"$@"''
+    }
```

(optional)

How about:

```suggestion
    exec ${if (isCross && crossSupport.canCheck) then
      (if stdenv.hostPlatform.isGhcjs then "node" else crossSupport.emulator)
    else
      ""} "$@"
```

(not sure what the formatter will do it with it)

This would `exec` in both cases, which might not be bad?